### PR TITLE
Update tari script cucumber tests to use standalone miner

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description

This PR requires PR #2833 
This PR ports the cucumber tests to use the standalone miner and not the build in cucumber miner.

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [ ] I'm merging against the `development` branch.
* [ ] I ran `cargo-fmt --all` before pushing.
* [ ] I ran `cargo test` successfully before submitting my PR.
* [ ] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: integration_tests/features/Propagation.feature [hunk 1]
```diff
@@ -4,8 +4,8 @@ Feature: Block Propagation
   Scenario Outline: Blocks are propagated through the network
     Given I have <NumSeeds> seed nodes
     And I have <NumNonSeeds> base nodes connected to all seed nodes
-    And I have a base node MINER connected to all seed nodes
-    When I mine <NumBlocks> blocks on MINER
+    And I have full mining package MINER connected to all seed nodes
```
Comment (stringhandler) [inline integration_tests/features/Propagation.feature]: This is not a good name. I am close to the source but I have no idea what this describes

File: integration_tests/features/Propagation.feature [hunk 2]
```diff
@@ -4,8 +4,8 @@ Feature: Block Propagation
...
+    And Mining node MINER mines <NumBlocks> blocks
```
Comment (stringhandler) [inline integration_tests/features/Propagation.feature]:
```suggestion
    And mining node MINER mines <NumBlocks> blocks
```

File: integration_tests/features/Propagation.feature [hunk 3]
```diff
@@ -60,34 +60,34 @@ Feature: Block Propagation
...
   Scenario: Nodes should never switch to block sync but keep insync via propagation
     Given I have 1 seed nodes
-    Given I have a base node MINER connected to all seed nodes
+    Given I have full mining package MINER connected to all seed nodes
     And I have a lagging delayed node LAG1 connected to node MINER with blocks_behind_before_considered_lagging 10000
     Given I have a lagging delayed node LAG2 connected to node MINER with blocks_behind_before_considered_lagging 10000
         # Wait for node to so start and get into listing mode
     When I wait 100 seconds
-    When I mine 5 blocks on MINER
+    Given Mining node MINER mines 5 blocks
```
Comment (stringhandler) [inline integration_tests/features/Propagation.feature]: This should be a `When`. `Given` is in the past tense and describes that something has been set up already. Either it should be `Given I have mined 5 blocks` or `When mining node MINER mines 5 blocks`

File: integration_tests/features/Propagation.feature (hunk 1 above)
Comment (SWvheerden) [reply integration_tests/features/Propagation.feature]: I agree, but its the best I could come up with. 
This creates a: wallet called MINER, a node called MINER, and a standalone miner called: MINER.

Comment (delta1) [reply integration_tests/features/Propagation.feature]: How about "I have an integrated mining setup" or something like that

Comment (stringhandler) [reply integration_tests/features/Propagation.feature]: Full mining package just sounds like it is inclusive of the merge mining proxy.

Comment (stringhandler) [reply integration_tests/features/Propagation.feature]: I would suggest:
"When I have SHA3 miner MINER connected to all seed nodes"
"And I mine 5 blocks on MINER"

The non-named base node and wallet can be random strings

Comment (delta1) [reply integration_tests/features/Propagation.feature]: It also sounds like Kubrick movie 😆

image

**Comment by stringhandler** [comment]
Is it possible to retarget this to the development branch?

File: integration_tests/features/support/steps.js [hunk 4]
```diff
@@ -303,6 +303,87 @@ Given(
   }
 );
 
+Given(
+  /I have a SHA3 miner (.*) connected to seed node (.*)/,  
+  { timeout: 40 * 1000 },
+  async function (name, seed) {
+    //add the base_node
+    const node = this.createNode(name);
```
Comment (stringhandler) [inline integration_tests/features/support/steps.js:311]: This duplication is not great, but I didn't see it in the first review. Hoping to add some clippy like stuff to our JS

**Review by stringhandler — APPROVED** [review]
Good enough to go in. Be careful of adding obviously duplicated code in future

**Pull request merged**